inc/class-registration.php

```diff
@@ -338,6 +338,13 @@ function ( $content ) {
 			}
 		);
 
+		add_action(
+			'wp_footer',
+			function() {
```

Comment:
Here won't this include the style on all pages, ignoring the previous conditions the previous conditions?

Reply:
No because if something has been enqueued already, it won't try to enqueue it again, we've conditions to make sure of that. This entire CSS enqueue is pretty ad-hoc as WordPress kept adding new things Widgets, Query, FSE and so on. While it works, it's not an ideal solution and there's a ticket to write CSS enqueue part from scratch to rethink the whole thing: https://github.com/Codeinwp/otter-blocks/issues/1081

Those different places take care of different scenarios, and one takes care for those that weren't already enqueued. Each is fail-safe for other. It's a mess but it works in all cases so far and nothings get enqueued twice.

Reply:
But that's for the entire existing solution, the wp_footer solution isn't ideal and we should find others way to deal with this than to use wp_footer.

Reply:
@Soare-Robert-Daniel We should use some action in Custom Layout to trigger the CSS as it might be used in various different ways.

Reply:
As if it's added to a hook or outside post context, our filter won't realize it's a post type.